src/Compiler/Driver/ReuseTcResults/ReuseTcResultsDriver.fs

```diff
@@ -0,0 +1,124 @@
+module internal FSharp.Compiler.ReuseTcResults
+
+open System
+open System.Collections.Generic
+open System.IO
+
+open FSharp.Compiler.CompilerConfig
+open FSharp.Compiler.Diagnostics
+open FSharp.Compiler.GraphChecking
+open FSharp.Compiler.IO
+open FSharp.Compiler.Syntax
+open FSharp.Compiler.Syntax.PrettyNaming
+
+open Internal.Utilities.Hashing
+
+let (++) a b = Path.Combine(a, b)
+
+[<Sealed>]
+type ReuseTcResultsDriver(tcConfig: TcConfig) =
+
+    let outputDir = tcConfig.outputDir |> Option.defaultValue ""
+    let tcDataFolderName = outputDir ++ FSharpTcDataResourceName
+
+    let cmdLineFilePath = tcDataFolderName ++ "cmdline"
+    let graphFilePath = tcDataFolderName ++ "graph"
+    let referencesFilePath = tcDataFolderName ++ "references"
+
+    let writeThisTcData path content =
```

Comment:
Some sort of checksum mechanism should also be incorporated into the file, so it's immediately apparent that both format is correct as well as file is well written.

Reply:
So I added a basic file structure:
```
BEGIN CMDLINE
...
cmd line
...
BEGIN GRAPH
...
graph
...
BEGIN REFERENCES
...
references
...
```
If, during reading, we find a header that's something else, we throw (we can handle it more gently later).

Do you think it's enough? If not, what particular checksum mechanism would you recommend?